### ISSUE TITLE
Screenreader should announce changes in selection for `SelectableRegion`

### ISSUE THREAD

**Issue by Renzo-Olivares** [flutter/flutter#182909]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

  1. Run any app with a SelectableRegion/SelectionArea that wraps a Text widget.
  2. Turn on screen reader.
  3. Attempt to select some text using screen reader, on Android this is talkback through selection mode, on iOS this is voiceover and text selection mode can be accessed through the rotor.

### Expected results
Text can be selected and when the selection changes what was selected or unselected is announced.

### Actual results
No text is selected/no selection is announced.

On Android this means double tapping with two fingers and holding does enable selection mode when the text has accessibility focus. Instead it announces "not selectable".

On iOS this means that the rotor does not show the "text selection" option when the text has accessibility focus.

On macOS this means as we select with mouse gestures nothing is announced.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(child: SelectionArea(child: Text('Hello, World!'))),
      ),
    );
  }
}
```
</details>
